### DISCUSSION TITLE
AddMedia Permission denied!

### DISCUSSION THREAD

**Discussion by ducdk** [spatie/laravel-medialibrary#2143] [Q&A]
When upload file image by api /upload of laravel-medialibrary and I want save file by function addMedia($pathToFile)->toMediaCollection('images');
But i can not addMedia because error
fopen(/var/www/upload.com/public/media/12/OKESTEw14xzwK7H5p86KXoIzC8QK9FmMEdOPccrH.jpeg): failed to open stream: Permission denied

I check in server centos when use api /upload file set permission 
![image](https://user-images.githubusercontent.com/20499340/97744736-55f2b080-1b1a-11eb-9339-386e25af63b2.png)
How to config permission file upload ?

Please help me! Thank iu

**Comment by yassermohammad** [comment]
Did u add the 'visibility' => 'public' to media in filesystems.php in the config like below?

        'media' => [
            'driver' => 'local',
            'root'   => public_path('media'),
            'url'    => env('APP_URL').'/media',
            'visibility' => 'public',
        ],
